dask/array/overlap.py

```diff
@@ -536,16 +577,30 @@ def coerce_depth(ndim, depth):
         depth = (depth,) * ndim
     if isinstance(depth, tuple):
         depth = dict(zip(range(ndim), depth))
-
+    if isinstance(depth,dict):
+        for i in range(ndim):
+            if i not in depth:
+                depth.update({i:0})
     return depth
 
 
 def coerce_boundary(ndim, boundary):
+    default = 'reflect'
     if boundary is None:
-        boundary = 'reflect'
+        boundary = default
     if not isinstance(boundary, (tuple, dict)):
         boundary = (boundary,) * ndim
     if isinstance(boundary, tuple):
         boundary = dict(zip(range(ndim), boundary))
-
+    if isinstance(boundary,dict):
+        for i in range(ndim):
+            if i not in boundary:
+                boundary.update({i:default})
     return boundary
+
+
+def neg_depth(x):
```

Comment:
The `neg_depth` function doesn't appear to be used anywhere. Is this functionality necessary?

Reply:
yes, I removed it, but haven't pushed